compaction/compaction_manager.cc

```diff
@@ -314,10 +315,11 @@ class compaction_manager::major_compaction_task : public compaction_manager::tas
         cmlog.info0("User initiated compaction started on behalf of {}.{}", t->schema()->ks_name(), t->schema()->cf_name());
         compaction_backlog_tracker bt(std::make_unique<user_initiated_backlog_tracker>(_cm._compaction_controller.backlog_of_shares(200), _cm._available_memory));
         _cm.register_backlog_tracker(bt);
-        // FIXME: co_await coroutine::switch_to
-        co_await with_scheduling_group(_cm._compaction_controller.sg(), [this, t, descriptor = std::move(descriptor)] () mutable {
-            return t->compact_sstables(std::move(descriptor), _compaction_data);
-        });
+
+        auto prev_sg = co_await coroutine::switch_to(_cm._compaction_controller.sg());
+        co_await t->compact_sstables(std::move(descriptor), _compaction_data);
+        co_await coroutine::switch_to(prev_sg);
```

Comment:
anyway, I realized that we restore the previous sg once we go out of the coroutine frame, minimizing the impact of a forgotten switch_to() call on previous sg.

Reply:
> what's preventing us from applying RAII for coroutine's sg switcher?

The problem is that to switch back we need to co_await coroutine::switch_to so it can reschedule the coroutine task with the restored scheduling group, but we can't co_await in the destructor of a RAII object.

Reply:
Simplified in the next draft version by switch_to covering the whole `run_to` coroutine, including both the compaction_manager::task level logic and the called compaction code.

The original scheduling group is restored automatically when the coroutine co_return:s